main.cpp

```diff
@@ -128,140 +178,205 @@ struct CarWash
         bool offlineManagement = false;
         float regularSalaryPerMonth = 0;
-        float percentageShareOfIncome = 30.0f;
+        bool ifTheShowIsCancelled = true;
+        int bookRehersalTime = 0;
         
         void bookRehersalRoom(float bandBudgetPerHour = 50);
-        void scheduleTheTourPlan(bool ifTheShowIsCancelled = true);
+        bool scheduleTheTourPlan(bool barOpen = false);
         void doMarketing(int posterPrintAmount = 100);
     };
 
     void doRehersal(bool allBandMemberAttended);
     void doLiveShow(int showTimeAmount = 120, int songNumbers = 25);
-    
-    //returns how much money the band is making
     float makeMoney(bool isThisShowHasTicket = true);
 };
 
-/*
-Thing 2) YouTuber
-5 properties:
-    1) hour amount on script
-    2) hour amount on shooting
-    3) hour amount on editing
-    4) view amount per video
-    5) money amount made from per 1000 view
-3 things it can do:
-    1) make video
-    2) increase follower
-    3) make money
-*/
+bool RockBand::BandManager::scheduleTheTourPlan(bool barOpen)
+{
+    if(barOpen)
+    {
+        bookRehersalTime -=1;
+    }
+    else
+    {
+        ifTheShowIsCancelled = true;
+    }
+    return ifTheShowIsCancelled;
+}
+
+void RockBand::BandManager::bookRehersalRoom(float bandBudgetPerHour)
+{
+    if(bandBudgetPerHour > 0 && ifTheShowIsCancelled == false)
+    {
+        bookRehersalTime += 1;
+    }
+}
+
+void RockBand::BandManager::doMarketing(int posterPrintAmount)
+{
+    if(ifTheShowIsCancelled)
+    {
+        posterPrintAmount = 0;   
+    }
+    else
+    {
+        posterPrintAmount = 300;   
+    }
+}
+
+void RockBand::doRehersal(bool allBandMemberAttended)
+{
+    if(allBandMemberAttended)
+    {
+        numberOfRehersalTimePerWeek +=1;
+    }
+}
+
+void RockBand::doLiveShow(int showTimeAmount, int songNumbers)
+{
+    songNumbers = showTimeAmount / 4;
+}
+
+float RockBand::makeMoney(bool isThisShowHasTicket)
+{
+    if(isThisShowHasTicket)
+    {
+        moneyMade = amountOfMoneyMadePerLiveShow * numberOfLiveShowTimePerWeek - amountOfMoneySpentPerRehersal * numberOfRehersalTimePerWeek;
+    }
+    return moneyMade;
+}
+
+
 struct YouTuber
 {
     float hourAmountOnScript = 4;
     float hourAmountOnShooting = 3;
     float hourAmountOnEditing = 3;
     int viewAmountPerVideo = 7000;
     float moneyAmountPerVideo = 10;
+    int videoNum = 1;
+    float moneyMade = 0.0f;
 
     void makeVideo(std::string videoTopic, RockBand ACDC, float timeLong = 8.0f);
     void increaseFollower(bool subNotify = true);
-    void makeMoney(bool monetization = true);
+    float makeMoney(bool monetization = true);
 };
 
+void YouTuber::makeVideo(std::string videoTopic, RockBand ACDC, float timeLong)
+{ 
+    timeLong = ACDC.numberOfPlayers * 2;
+    videoTopic = "ACDC";
+}
+
+void YouTuber::increaseFollower(bool subNotify)
+{
+    if(subNotify)
+    {
+        int follower;
+        follower = (int) round(viewAmountPerVideo * 0.01);
```

Comment:
Casting has not been covered yet. Don't do any casting here. Change your code so that it doesn't produce a warning regarding type conversion. 
Also don't declare a variable and then initializing on the next line.
just do it all on the same line.
- [ ] revise